src/js/helpers/DataPanel.ts

```diff
@@ -44,152 +17,150 @@ function esriQuery(url: string, queryParams: any): Promise<__esri.FeatureSet> {
-  map: Map | undefined,
-  mapview: MapView,
-  mapPoint: Point,
-  layerFeatureResults: LayerFeatureResult[]
-): Promise<any> {
-  if (map) {
-    const processedLayerInfo = layerFeatureResults.map(f => f.layerID);
-    const visibleServerLayers = map.layers
-      .filter(l => l.visible && l.type !== 'graphics')
-      .filter(l => !processedLayerInfo.includes(l.id));
-    const processedLayers = await processLayers(
-      mapview,
-      mapPoint,
-      visibleServerLayers
-    );
-    return processedLayers;
-  }
+  return processedLayerResult;
 }
 
 //Client Side Feature Fetching
-export function addPopupWatchUtils(
+export async function queryLayersForFeatures(
   mapview: MapView,
   map: Map | undefined,
-  mapPoint: Point
-): void {
-  //Watching for promises array on Popup, that gets resolved once user clicks on the map and features are returned. Note: this will not return any server side features.
-  once(mapview.popup, 'promises', promises => {
-    function resolveClientPromisesWithErrors(promises: any): Promise<void> {
-      return Promise.all<any>(
-        promises.map((p: Promise<any>) => p.catch((error: Error) => null))
-      ).then(async clientFeatures => {
-        let layerFeatureResults: LayerFeatureResult[] = [];
-        const cleanClientFeautures = clientFeatures
-          .filter(f => f !== null) //catch failed promises
-          .filter(f => f.length !== 0) //catch no features returned
-          .map((featureObject: Graphic[]) => {
-            //extract layerID and layerTitle from resulting feature object
-            const {
-              layerID,
-              layerTitle,
-              sublayerID,
-              sublayerTitle
-            } = extractLayerInfo(featureObject);
-            const newFeatureObject: LayerFeatureResult = {
-              layerID: layerID,
-              layerTitle: layerTitle,
-              sublayerID: sublayerID,
-              sublayerTitle: sublayerTitle,
-              features: featureObject.map(g => {
+  event: __esri.MapViewClickEvent
+): Promise<void> {
+  let layerFeatureResults: LayerFeatureResult[] = [];
+
+  const queryParams = {
+    where: '1=1',
+    outFields: ['*'],
+    units: 'miles',
+    distance: 0.02 * mapview.resolution,
+    geometry: event.mapPoint,
+    returnGeometry: true
+  };
+
+  const allLayersVisibleLayers: any = map?.layers
+    .filter(l => l.visible && l.type !== 'graphics')
+    .toArray();
+
+  if (allLayersVisibleLayers) {
+    for await (const layer of allLayersVisibleLayers) {
+      // Deal with CLIENT side layers
+      //TODO: is there a better way to split client/server queries?
+      if (
+        layer.type === 'feature' ||
+        layer.type === 'csv' ||
+        layer.type === 'geojson' ||
+        layer.type === 'scene'
+      ) {
+        try {
+          const featureResults = await layer.queryFeatures(queryParams);
+          //Ignore empty results
+          if (featureResults.features.length !== 0) {
```

Comment:
affirmative check? Could help us ID edge-cases later on (if `features` is undefined, or is an unexpected result, for example)
`if (featureResults.features.length > 0) {`